app/lib/features/pins/pages/pin_details_page.dart

```diff
@@ -0,0 +1,286 @@
+import 'package:acter/common/providers/common_providers.dart';
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/widgets/edit_html_description_sheet.dart';
+import 'package:acter/common/widgets/edit_title_sheet.dart';
+import 'package:acter/common/widgets/render_html.dart';
+import 'package:acter/features/attachments/widgets/attachment_section.dart';
+import 'package:acter/features/comments/widgets/comments_section.dart';
+import 'package:acter/features/home/widgets/space_chip.dart';
+import 'package:acter/features/pins/Utils/pins_utils.dart';
+import 'package:acter/features/pins/actions/edit_pin_actions.dart';
+import 'package:acter/features/pins/actions/pin_link_backward_support.dart';
+import 'package:acter/features/pins/actions/reduct_pin_action.dart';
+import 'package:acter/features/pins/actions/report_pin_action.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:skeletonizer/skeletonizer.dart';
+
+class PinDetailsPage extends ConsumerStatefulWidget {
+  static const pinPageKey = Key('pin-page');
+  static const actionMenuKey = Key('pin-action-menu');
+  static const editBtnKey = Key('pin-edit-btn');
+  static const titleFieldKey = Key('edit-pin-title-field');
+  static const descriptionFieldKey = Key('edit-pin-description-field');
+
+  final String pinId;
+
+  // ignore: use_key_in_widget_constructors
+  const PinDetailsPage({
+    Key key = pinPageKey,
+    required this.pinId,
+  });
+
+  @override
+  ConsumerState<PinDetailsPage> createState() => _PinDetailsPageState();
+}
+
+class _PinDetailsPageState extends ConsumerState<PinDetailsPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(),
+      body: _buildBodyUI(),
+    );
+  }
+
+  AppBar _buildAppBar() {
+    return AppBar(
+      actions: [_buildActionMenu()],
+    );
+  }
+
+  Widget _buildBodyUI() {
+    final pinData = ref.watch(pinProvider(widget.pinId));
+    return pinData.when(
+      data: (pin) {
+        manageBackwardPinLinkSupport(context, ref, pin);
```

Comment:
yeah, we can't do it this way.

1) this is called on every render ... regardless of whether we already tried
2) if we don't have the permission, it doesn't actually reset the pin, so we'd be resubmitting this over and over again
3) if we don't have permission to submit attachments, we are spamming the space with bad events..


What I'd be doing instead is this:
1. if have a link, we put a "fake" (looks like) attachment in the UI that is purely for backwards compat
2. if the user edits it, we remove the existing and submit as a new attachment.
3. (as we don't have a link anymore, the fake attachment disappears)

Let's make this an explicit conversion upon interaction, not an implicit one that could fail and have bad failure scenarios.

Reply:
Thanks @gnunicorn for noticing glitches. I have did required changes as your suggestion in [this commit.](https://github.com/acterglobal/a3/pull/2067#issuecomment-2294911121). Also shared video recording for better visualisation. :)